### PR TITLE
Tests for StimulusManager and refactor internal stimuli data structure

### PR DESCRIPTION
**Pull request by WeinaJi**
## Context
For #249, this PR adds the unit tests for `StimulusManager` class. 
It also refactors the internal data structures for stimuli: removes the legacy `parsedInjects` and uses `parsedStimuli` instead.

## Scope
- Refactor in `sonata_config.py`,`configurations.py` and `node.py`

## Testing
- new file `tests/unit/test_stimulus_manager.py`

## Review
* [x] PR description is complete
* [x] Coding style (imports, function length, New functions, classes or files) are good
* [x] Unit/Scientific test added
* [ ] Updated Readme, in-code, developer documentation

### CODE REVIEW

File: neurodamus/io/sonata_config.py [hunk 1]
```diff
@@ -361,7 +359,7 @@ def parsedConnects(self):
         }
 
     @property
-    def parsedStimuli(self):
+    def parsedStimuli(self) -> list:
```
Comment (cattabiani) [inline neurodamus/io/sonata_config.py:362]: docstring is missing and the function is not completely trivial

File: neurodamus/io/sonata_config.py [hunk 2]
```diff
@@ -371,28 +369,21 @@ def parsedStimuli(self):
         }
         module_translation = {"seclamp": "SEClamp", "subthreshold": "SubThreshold"}
 
-        stimuli = {}
-        for name in self._sim_conf.list_input_names:
+        stimuli = []
+        # TODO: loop over self._sim_conf.list_input_names,
```
Comment (cattabiani) [inline neurodamus/io/sonata_config.py:373]: do you still have this to do?

File: neurodamus/io/sonata_config.py [hunk 3]
```diff
@@ -371,28 +369,21 @@ def parsedStimuli(self):
...
+        # The order of stimulus injection could lead to minor difference on the results
+        # so need to preserve it as in the config file, libsonata API
+        for name in self._sections["inputs"]:
             stimulus = self._translate_dict("inputs", self._sim_conf.input(name))
```
Comment (cattabiani) [inline neurodamus/io/sonata_config.py:377]: while you are at it, may you add a docstring at this function too?

File: tests/unit/test_stimulus_manager.py [hunk 4]
```diff
@@ -0,0 +1,881 @@
...
+        "Delay": 0,
+        "Dt": 0.5,
+        "Tau": 2.7,
+        "Reversal": 0.1,
+        "MeanPercent": -10,
+        "SDPercent": 1,
+    }
+    with caplog.at_level(logging.WARNING):
+        ringtest_stimulus_manager.interpret(target_onecell, stim_info)
+        assert "RelativeOrnsteinUhlenbeck signal is mostly zero" in caplog.text
```
Comment (cattabiani) [inline tests/unit/test_stimulus_manager.py:881]: I know it is almost trivial but yous aid previously that the order of registering of the signals may be important. May you add a test where you check that the registering order is preserved against (with corner cases if they are there)

**Review by cattabiani — COMMENTED** [review]
almost LGTM

File: neurodamus/core/configuration.py [hunk 5]
```diff
@@ -612,9 +603,9 @@ def _stimulus_params(config: _SimConfig):
         },
     }
     deprecated_values = {"Pattern": ("NPoisson", "NPoissonInhomogeneous", "ReplayVoltageTrace")}
-    for name, stim in config.stimuli.items():
+    for stim in config.stimuli:
         _check_params(
-            "Stimulus " + name,
+            "Stimulus " + stim["Name"],
```
[the remark this answers is not in the document]
Comment (WeinaJi) [reply neurodamus/core/configuration.py:608]: I've changed config.stimuli to a list instead of a dict. So it is clear that the order is preserved. We could eventually do the same for connections, modifications. cc @mgeplf

File: neurodamus/io/sonata_config.py (hunk 2 above)
Comment (WeinaJi) [reply neurodamus/io/sonata_config.py:373]: yes, after we update the SONATA spec and libsonata API

File: neurodamus/core/configuration.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (cattabiani) [reply neurodamus/core/configuration.py:608]: ok. resolvable for me. I leave it open for Mike

File: neurodamus/io/sonata_config.py (hunk 3 above)
Comment (WeinaJi) [reply neurodamus/io/sonata_config.py:377]: which function do you mean in this comment?

File: tests/unit/test_stimulus_manager.py (hunk 4 above)
Comment (WeinaJi) [reply tests/unit/test_stimulus_manager.py:881]: Which order? The order of `SimConfig.stimuli` which determines the order of `self._stim_manager.interpret(target_spec, stim)` in node.py? There is one in `test_sonata_confg.py::test_parse_inputs`

File: neurodamus/io/sonata_config.py (hunk 3 above)
Comment (cattabiani) [reply neurodamus/io/sonata_config.py:380]: `_translate_dict`

Comment (WeinaJi) [reply neurodamus/io/sonata_config.py:380]: Done

File: neurodamus/io/sonata_config.py (hunk 1 above)
Comment (WeinaJi) [reply neurodamus/io/sonata_config.py:362]: Done

**Review by mgeplf — APPROVED** [review]

**Review by cattabiani — APPROVED** [review]
LGTM
